packages/artifact/src/internal/upload-http-client.ts

```diff
@@ -406,14 +407,18 @@ export class UploadHttpClient {
     isGzip: boolean,
     totalFileSize: number
   ): Promise<boolean> {
+    // open a new stream and read it to compute the digest
+    const digest = await digestForStream(openStream())
```

Comment:
note: we'll have to read the data chunk twice to calculate the CRC digest. Not sure if there is a more clever solution where we can calculate during chunking (or if it's worth it 🤷)

cc @konradpabjan

Reply:
My initial hunch is this shouldn't be a concern. The majority of the time as part of artifact upload is due to networking so a small overhead to compute the digest should be fine.